src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -231,6 +237,13 @@ export default class ComboBox extends React.PureComponent<Props, State> {
       <div ref={this.popoverScrollContainer} />
     );
 
+    const emptyStateMarkup = !actionsBefore &&
+      !actionsAfter &&
+      options.length === 0 &&
```

Comment:
What are the pitfalls to this approach? We use it all the times for string

Reply:
On second thought, that statement was too strong. I think we should avoid type coercion because this can result in unexpected cases, but in this case it is technically safe. Although, I think the main benefit of avoiding type coercion in a condition is for readability. We have this rule outlined [here](https://github.com/shopify/javascript#11.5).